### PR TITLE
Add context, examples, and images to Snaps custom UI

### LINKED ISSUE #843: Improve "Custom UI" section of Snaps documentation
AC:

- Custom UI sample code is put in context, i.e. as it would be used with `snap_dialog` or `onTransaction` return value

### LINKED ISSUE #890: Snaps page for Custom UI should have images
Because this is a visual thing we should show not just tell: https://docs.metamask.io/snaps/how-to/use-custom-ui/

### PR DESCRIPTION
**Pull request by alexandratran**
Improve Snaps custom UI examples to include usage with `snap_dialog`, `onTransaction`, or `onHomePage`, and include screenshots of all UI components. Minor fixes to images across the site for consistency.

Preview: https://docs.metamask.io/843-improve-custom-ui/snaps/features/custom-ui/

Fixes #843 
Fixes #890

### CODE REVIEW

File: snaps/features/custom-ui.md [hunk 1]
```diff
@@ -6,206 +6,365 @@ sidebar_position: 1
...
+import { panel, heading, text } from "@metamask/snaps-sdk";
+
+await snap.request({
+    method: "snap_dialog",
+    params: {
+        type: "alert",
+        const content = panel([
+            heading("Alert heading"),
+            text("Something happened in the system."),
+        ]);
```
Comment (Mrtenz) [inline snaps/features/custom-ui.md:37]:
```suggestion
        content: panel([
            heading("Alert heading"),
            text("Something happened in the system."),
        ]),
```

File: snaps/features/custom-ui.md [hunk 2]
```diff
@@ -6,206 +6,365 @@ sidebar_position: 1
...
+import { panel, heading, address } from "@metamask/snaps-sdk";
+
+await snap.request({
+    method: "snap_dialog",
+    params: {
+        type: "alert",
+        const content = panel([
+            heading("Are you sure you want to send tokens to this address?"),
+            address("0x000000000000000000000000000000000000dEaD"),
+        ]);
```
Comment (Mrtenz) [inline snaps/features/custom-ui.md:64]:
```suggestion
       content: panel([
            heading("Are you sure you want to send tokens to this address?"),
            address("0x000000000000000000000000000000000000dEaD"),
        ]),
```

File: snaps/features/custom-ui.md [hunk 3]
```diff
@@ -6,206 +6,365 @@ sidebar_position: 1
...
+import { text, copyable } from "@metamask/snaps-sdk";
+
+await snap.request({
+    method: "snap_dialog",
+    params: {
+        type: "alert",
+        const content = panel([
+            text("Your address:"),
+            copyable("0x000000000000000000000000000000000000dEaD"),
+        ]);
```
Comment (Mrtenz) [inline snaps/features/custom-ui.md:94]:
```suggestion
        content: panel([
            text("Your address:"),
            copyable("0x000000000000000000000000000000000000dEaD"),
        ]),
```

File: snaps/features/custom-ui.md [hunk 4]
```diff
@@ -6,206 +6,365 @@ sidebar_position: 1
...
+import { panel, row, text, address } from "@metamask/snaps-sdk";
+
+await snap.request({
+    method: "snap_dialog",
+    params: {
+        type: "alert",
+        const content = panel([
+            row("Address", address("0x000000000000000000000000000000000000dEaD")),
+            row("Balance", text("1.78 ETH")),
+        ]);
```
Comment (Mrtenz) [inline snaps/features/custom-ui.md:230]:
```suggestion
        content: panel([
            row("Address", address("0x000000000000000000000000000000000000dEaD")),
            row("Balance", text("1.78 ETH")),
        ]),
```

File: snaps/features/custom-ui.md [hunk 5]
```diff
@@ -6,206 +6,365 @@ sidebar_position: 1
...
+import { panel, heading, spinner } from "@metamask/snaps-sdk";
+
+await snap.request({
+    method: "snap_dialog",
+    params: {
+        type: "alert",
+        const content = panel([
+            heading("Please wait..."),
+            spinner(),
+        ]);
```
Comment (Mrtenz) [inline snaps/features/custom-ui.md:255]:
```suggestion
        content: panel([
            heading("Please wait..."),
            spinner(),
        ]),
```

File: snaps/features/custom-ui.md [hunk 6]
```diff
@@ -6,206 +6,365 @@ sidebar_position: 1
...
+import { panel, heading, text } from "@metamask/snaps-sdk";
+
+await snap.request({
+    method: "snap_dialog",
+    params: {
+        type: "alert",
+        const content = panel([
+            heading("Hello world!"),
+            text("This is **bold** and this is _italic_."),
+        ]);
```
Comment (Mrtenz) [inline snaps/features/custom-ui.md:305]:
```suggestion
        content: panel([
            heading("Hello world!"),
            text("This is **bold** and this is _italic_."),
        ]),
```

**Review by Mrtenz — CHANGES_REQUESTED** [review]
This looks great! Just found some issues in the examples.

**Review by Mrtenz — APPROVED** [review]

File: wallet/how-to/manage-permissions.md [hunk 7]
```diff
@@ -21,10 +21,10 @@ the [`eth_accounts`](/wallet/reference/eth_accounts) restricted method:
 
 <div class="row margin-bottom--md">
     <div class="column">
-        <img src={require("../assets/request-permissions.png").default} alt="Request permissions confirmation 1" style={{border: '1px solid gray'}} />
+        <img src={require("../assets/request-permissions.png").default} alt="Request permissions confirmation 1" style={{border: '1px solid #DCDCDC'}} />
     </div>
     <div class="column">
-        <img src={require("../assets/request-permissions-2.png").default} alt="Request permissions confirmation 2" style={{border: '1px solid gray'}} />
+        <img src={require("../assets/request-permissions-2.png").default} alt="Request permissions confirmation 2" style={{border: '1px solid #DCDCDC'}} />
```
Comment (joaniefromtheblock) [inline wallet/how-to/manage-permissions.md:27]: these images look a little blurry

**Review by joaniefromtheblock — APPROVED** [review]
LGTM

**Pull request merged**